src/sinks/datadog/metrics/request_builder.rs

```diff
@@ -332,19 +278,22 @@ fn encode_now_or_never(
 
     encoder
         .finish()
-        .map(|(payload, mut processed, raw_bytes_written)| {
-            let json_size = processed.estimated_json_encoded_size_of();
+        .map(|(payload, mut processed, uncompressed_payload_len)| {
             let finalizers = processed.take_finalizers();
             let ddmetrics_metadata = DDMetricsMetadata {
                 api_key,
                 endpoint,
                 finalizers,
-                raw_bytes: raw_bytes_written,
             };
-            let builder = RequestMetadataBuilder::new(metrics_len, raw_bytes_written, json_size);
-            let bytes_len =
-                NonZeroUsize::new(payload.len()).expect("payload should never be zero length");
-            let request_metadata = builder.with_request_size(bytes_len);
+
+            let request_metadata = RequestMetadata::new(
+                processed.len(),
+                processed.size_of(),
+                uncompressed_payload_len,
+                payload.len(),
+                processed.estimated_json_encoded_size_of(),
+            );
+            debug!("dd metrics request metadata: {:?}", request_metadata);
```

Comment:
If this is remaining shouldn't it follow our spec/style for log messages?

Reply:
Woops. No, this should definitely get removed.

Reply:
Fixed in a3c8227c6.